### ISSUE TITLE
Cache on disc has a race condition

### ISSUE THREAD

**Issue by davidanthoff** [JuliaEditorSupport/LanguageServer.jl#37]
The biggest problem I think is still the on disc cache. I think at the end of the day we have a massive race condition there, as soon as someone is using two instances of vscode at the same time I think that can trigger, right? So my preference would still be that we do away with that, and instead build the cache anew in every instance of the LS and only in-memory. 

With #36 it takes about 10 seconds on my system to extract all the information from base, and I assume it should be much faster to extract stuff from packages, so maybe we can just say the cache is rebuilt in-memory for every LS instance?

I guess one problem is how to move the doc info from the child process to the LS. Maybe we could serialize things, then encode them as text and just stream them via STDOUT to master? Or another idea would be to create a temporary file (with a temporary filename) that is used to transfer just one set of results back to the LS process, but we never write anything merged out to disc.

Another idea might be to actually ``spawn`` the process that does all the work, and let the LS continue to run while the child process runs. Once the child process is done, its results could be merged into the in memory cache. This would mean that the LS could still respond to messages while the child process extracts stuff. If a request for a doc comes in that hasn't been added to the cache, it would just return empty, which still seems better than just blocking the LS?

**Comment by ZacLN** [comment]
I'm all for this, I think the `spawn` approach would be best as dead time on startup is a bit of a pain. I'll have a look into this today

**Comment by ZacLN** [comment]
ref #38
